### PR TITLE
add iETH-FLI-P deploy script

### PR DESCRIPTION
**Pull request by gmridul**
Since MATIC FLI deploy scripts are blocked on new parameters, going ahead with this one first. 

Will rebase #101 on top of this.

### CODE REVIEW

File: polygon/deployments/constants/001_iethfli_system.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+
+export const CONTRACT_SETTINGS = {
+  COLLATERAL_DECIMAL_ADJUSTMENT: BigNumber.from(22),      // Decimal adjustment for chainlink. Equal to 28-decimals (10^18 * 10^18 / 10^dec / 10^8)
+  BORROW_DECIMAL_ADJUSTMENT: BigNumber.from(10),          // Decimal adjustment for chainlink. Equal to 28-decimals (10^18 * 10^18 / 10^dec / 10^8)
+};
+
+export const TOKEN = {
+  NAME: "iETH Flexible Leverage Index",
```
Comment (ncitron) [inline polygon/deployments/constants/001_iethfli_system.ts:28]: Have you confirmed that this is the correct name and symbol that they want?

File: polygon/deploy/001_iethfli_system.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  );
+
+  await setOperator(hre, CONTRACT_NAMES.BASE_MANAGER_NAME, await findDependency(EWG_MULTI_SIG));
+  await setManager(hre, CONTRACT_NAMES.IETHFLI, await getContractAddress(CONTRACT_NAMES.BASE_MANAGER_NAME));
+
+  //
+  // Helper Functions
+  //
+  async function polyfillForDevelopment(): Promise<void> {
+    if (await findDependency(SET_TOKEN_CREATOR) === "") {
```
[the remark this answers is not in the document]
Comment (ncitron) [reply polygon/deploy/001_iethfli_system.ts:175]: Same goes for a lot of the helper functions for initializing modules.

**Review by ncitron — COMMENTED** [review]
Looking good. Feel free to go ahead and deploy to staging mainnet. Once we have that I can setup some bots and we can lever it up.

File: polygon/deploy/001_iethfli_system.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmridul) [reply polygon/deploy/001_iethfli_system.ts:175]: yeah, this script was supposed to go after MATIC scripts, and didn't have these polyfills. While rebasing, I'll remove redundant code.

**Pull request merged**